src/test/java/com/aws/greengrass/clientdevices/auth/helpers/TestHelpers.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.helpers;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * Helpers for generic assertion tooling
+ */
+public final class TestHelpers {
+    public static final long DEFAULT_GENERIC_POLLING_TIMEOUT_MILLIS =
+            TimeUnit.SECONDS.toMillis(10);
+
+    private TestHelpers() {
+    }
+
+    public static boolean eventuallyTrue(Supplier<Boolean> condition, long... optional) throws InterruptedException {
```

Comment:
why not overload instead of varargs?  would still allow for optional args and have the benefits of arg names